### ISSUE TITLE
OGS no Demoboard feature

### ISSUE THREAD

**Issue by wizerdo** [CmdrDats/igoki#106]
What is was hoping for, is that you were able to set up a demo board (so that you input both players) is this at all possible? 
A relevant use case would be for recording games during tournaments. 

Hope to hear from you.

**Comment by CmdrDats** [comment]
Ye, the simplest base case records both players - it looks like there's a small bug at the moment where you need to hit file -> new sgf after you set camera up for it to start recording the game

**Comment by wizerdo** [comment]
You mean that you have to setup the program twice, and run it twice?

**Comment by CmdrDats** [comment]
No, simply setup the camera (select camera, choose board corners), put a few stones down to check that it's picking them up ok, take them off - then go to file -> new, and start playing - it should start recording the game. If you want to start with handicap stones, place the handicap stones before the file -> new, and it'll initialise the board to that position and start playing. I think it should also then start with white, though you'll need to check.

Let me know if that answers the question or if I'm misunderstanding perhaps?

**Comment by wizerdo** [comment]
Thx for the response, but indeed think you misunderstand. OGS has a feature that is called demo board, with it you can review games, or relay games that are played in a tournament. With it you play as both black and white. 

As I understand, Igoki currently only lets you play one color when connected to ogs, and demo boards don’t show up in Igoki

**Comment by CmdrDats** [comment]
Ah yes, i see. I'll have to figure out integration for demo board. For now, i think you can workaround this using the 'manual' tab instead of direct ogs integration. This basically simulates mouse clicks, so you can open demo board on one half of the screen and have it input both colours as mouse clicks. It also lets you integrate with just about any go client..

Let me know if that works as a workaround for now?

**Comment by wizerdo** [comment]
Yeah that will work I guess, I was also considering running the application twice and letting them play as black or white (against each other so to speak).

**Comment by CmdrDats** [comment]
I'll have a look tonight and see what it'll take to implement the demo board, i suspect it won't be too difficult to make it play for both

**Comment by wizerdo** [comment]
the manual input seems (at least for me) only to hover over the correct coordinate, but it doesn't place a move. (in OGS i have single click to confirm move enabled.

**Comment by CmdrDats** [comment]
One thing I've noticed, in some OSes, the first click takes window focus if not focused, and doesn't register click, so perhaps check that the browser window is focused? Else maybe i can make it double click, that might help

**Comment by wizerdo** [comment]
I did “focus” the window so don’t think that was it, but double click might work 🤔

See the images on how I have set it up for testing 😉

![1948B660-04E2-448C-894F-1EE40C97F73F](https://user-images.githubusercontent.com/94693927/143688320-eaa6be0d-8a54-4abe-8af9-910c5527a92b.jpeg)
![D7486727-A353-449A-9E97-90A30F60CDCB](https://user-images.githubusercontent.com/94693927/143688323-50bc3766-2955-4cb8-b616-3db733a15ea7.jpeg)

**Comment by CmdrDats** [comment]
Woo! That's a gorgeous setup!

I'll try change it to double click and maybe make the clicks slower and we can see.

**Comment by CmdrDats** [comment]
btw - I just checked my code - it should move the mouse back to original location after mouse click on manual mode - if it's just staying there, I suspect some other kind of error - if you've got a terminal window with output, could you check if there's some kind of error message there?

I've implemented a double click, and I'm going to test the OGS demo board soon and see if I can implement it properly.

**Comment by wizerdo** [comment]
27 nov. 2021 16:02:39,151 INFO  igoki.integration.robot - Submitting mouse click at:  pp
size:  19
cell:  [33 33]
coords:  [15 15]
ref: 
m [551 737]
0
1

this it was it says, doesn't seem like an error message

**Comment by wizerdo** [comment]
I was wondering, would it be possible to login into 2 separate accounts in Igoki, and “let them play against each other”

**Comment by CmdrDats** [comment]
hmm, interesting idea - it's entirely possible, sure.  I would need to rework a few things just to make the state management cleaner, but there's no fundamental reason why not.

What's your specific use case? Do you often have OGS players that can login with their details and then play each other while people can watch on OGS?

**Comment by wizerdo** [comment]
I think the use case would be live streaming tournaments, in such a manner that someone can comment on the game. The main problem with for example using a demo board on OGS is, that if you hire a pro for commentary on the games, then they cant properly review the game on OGS. so therefore one would like to play with 2 accounts, placing alternating moves on OGS, like a regular game.

**Comment by CmdrDats** [comment]
cool, makes sense - I had a look at OGS - the documentation on the demo/reviews are sparse, though I can probably make it work.. feels like two OGS accounts might be simpler and possibly more versatile?
